x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -133,7 +84,22 @@ export async function countInactiveAgents(
   return total;
 }
 
-export async function getAgent(soClient: SavedObjectsClientContract, agentId: string) {
+export async function getAgent(
+  soClient: SavedObjectsClientContract,
+  esClient: ElasticsearchClient,
+  agentId: string
+) {
+  const fleetServerEnabled = appContextService.getConfig()?.agents?.fleetServerEnabled;
+
+  if (fleetServerEnabled) {
+    const agentHit = await esClient.get<ESSearchHit<AgentSOAttributes>>({
```

Comment:
I'd expect this to use `crudServiceFleetServer.getAgent`?

Reply:
Yes it should, I am going to fix that :)